core/src/main/java/org/apache/iceberg/rest/ResourcePaths.java

```diff
@@ -63,4 +63,8 @@ public String table(TableIdentifier ident) {
         "v1", prefix, "namespaces", RESTUtil.encodeNamespace(ident.namespace()), "tables",
         RESTUtil.encodeString(ident.name()));
   }
+
+  public String rename() {
+    return SLASH.join("v1", prefix, "tables", "rename");
```

Comment:
The path was hard-coded for rename. I'm adding this here since this PR already modifies all the other lines to add headers. FYI @kbendick.